frontend/src/cards/MapCard.tsx

```diff
@@ -48,100 +49,71 @@ function MapCardWithKey(props: MapCardProps) {
+    if (
+      props.currentBreakdown === possibleBreakdown ||
+      props.currentBreakdown === "all"
+    ) {
+      queries[possibleBreakdown] = new MetricQuery(
+        props.metricConfig.metricId,
+        Breakdowns.byState().addBreakdown(
+          possibleBreakdown as BreakdownVar,
+          props.nonstandardizedRace
+        )
+      );
     }
-  }
-  if (["race_and_ethnicity", "all"].includes(props.currentBreakdown)) {
-    const raceQuery = new MetricQuery(
-      props.metricConfig.metricId,
-      Breakdowns.byState().andRace(props.nonstandardizedRace)
-    );
-    queries.push(raceQuery);
-    // If all are enabled, race should be the default inital query
-    initalQuery = raceQuery;
-  }
+  });
 
   return (
     <CardWrapper
-      queries={queries}
+      queries={Object.values(queries) as MetricQuery[]}
       datasetIds={getDependentDatasets([props.metricConfig.metricId])}
       titleText={`${
         props.metricConfig.fullCardTitleName
       } in ${props.fips.getFullDisplayName()}`}
     >
       {() => {
-        const queryResponse = datasetStore.getMetrics(initalQuery);
-        let mapData = queryResponse.data
-          .filter((row) => row.race_and_ethnicity !== "Not Hispanic or Latino")
-          .filter(
-            (r) =>
-              r[props.metricConfig.metricId] !== undefined &&
-              r[props.metricConfig.metricId] !== null
-          );
+        const currentlyDisplayedBreakdown: BreakdownVar =
+          props.currentBreakdown === "all"
+            ? "race_and_ethnicity"
+            : props.currentBreakdown;
+        const queryResponse = datasetStore.getMetrics(
+          queries[currentlyDisplayedBreakdown]
+        );
+        const breakdownValues = queryResponse.getUniqueFieldValues(
+          currentlyDisplayedBreakdown
+        );
+        if (breakdownFilter === "") {
+          setBreakdownFilter(breakdownValues[0]);
+        }
+
+        let filters: Array<(row: Row) => boolean> = [
+          (row) => row.race_and_ethnicity !== "Not Hispanic or Latino",
+          (row) => row[props.metricConfig.metricId] !== undefined,
+          (row) => row[props.metricConfig.metricId] !== null,
+        ];
         if (!props.fips.isUsa()) {
           // TODO - this doesn't consider county level data
-          mapData = mapData.filter((r) => r.state_fips === props.fips.code);
+          filters.push((row: Row) => row.state_fips === props.fips.code);
         }
         if (props.enableFilter) {
-          mapData = mapData.filter(
-            (r) => r.race_and_ethnicity === breakdownFilter
+          filters.push(
+            (row: Row) => row[currentlyDisplayedBreakdown] === breakdownFilter
           );
         }
 
+        // Remove any row for which we find a filter that returns false.
+        const mapData = queryResponse.data.filter(
+          (row: Row) =>
+            filters.find((filter) => filter(row) === false) === undefined
```

Comment:
I think this would be more readable if you switched to use Array.every(). Also maybe rename filters to filterConditions (or filterPredicates?):
```
(row: Row) => filterConditions.every((condition) => condition(row))
```